packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -0,0 +1,123 @@
+ * Driver plugin that serves as the link between the Sayt data source
+ * and the view layer. This plugin is responsible for listening to events,
+ * sending search requests to internal data source APIs,
+ * and then emitting the API response back in an event.
+ */
+export default class SaytDriverPlugin implements Plugin {
+  get metadata(): PluginMetadata {
+    return {
+      name: 'sayt_driver',
+      depends: [
+        this.eventsPluginName,
+        'sayt',
+      ],
+    };
+  }
+
+  /**
+   * Holds reference to the plugin registry that is received in registration.
+   */
+  core: PluginRegistry;
+  /**
+   * Name of the events plugin.
+   */
+  eventsPluginName: string = 'dom_events';
+  /**
+   * Event names for listening and dispatching.
+   */
+  saytDataEvent: string = 'sfx::autocomplete_fetch_data';
+  saytResponseEvent: string = 'sfx::autocomplete_received_results';
+  saytErrorEvent: string = 'sfx::autocomplete_sayt_error';
+
+  constructor() {
+    this.fetchSaytData = this.fetchSaytData.bind(this);
+  }
+
+  /**
+   * Callback method for when the plugin is registered in Core.
+   * The method will set the plugin registry to the plugin's internal 'core' property.
+   *
+   * @param plugins The plugin registry object from Core.
+   */
+  register(plugins: PluginRegistry): void {
+    this.core = plugins;
+  }
+
+  /**
+   * Lifecycle event where the plugin can first safely interact with the registry.
+   * The method will register an event listener for listening to Sayt data requests
+   * and trigger its callback for making search requests.
+   */
+  ready(): void {
+    this.core[this.eventsPluginName].registerListener(this.saytDataEvent, this.fetchSaytData);
+  }
+
+  /**
+   * Lifecycle event where the plugin will unregister all event listeners.
+   */
+  unregister(): void {
+    this.core[this.eventsPluginName].unregisterListener(this.saytDataEvent, this.fetchSaytData);
+  }
+
+  /**
+   * Callback for the Sayt data request event listener.
+   * Dispatches an event with the response from the sayt data plugin.
+   *
+   * @param saytDataQuery Request object received from the event listener.
+   */
+  fetchSaytData(saytDataQuery: SaytDataPayload): void {
+    const response = this.sendSaytAPIRequest(saytDataQuery);
+    response.then((data: any) => {
+      this.core[this.eventsPluginName].dispatchEvent(this.saytResponseEvent, data);
+    })
+    .catch((e) => {
+      this.core[this.eventsPluginName].dispatchEvent(this.saytErrorEvent, e);
+    });
+  }
+
+  /**
+   * Splits the request payload into the query string and a config object.
+   * Sends a search request to the Sayt data plugin with a callback to reformat.
+   *
+   * @param saytDataQuery Request object received from the event listener.
+   * @returns A promise from the Sayt API that has been reformatted
+   * with the passed callback.
+   */
+  sendSaytAPIRequest(saytDataQuery: SaytDataPayload): Promise<string[]> {
+    const { query, ...config } = saytDataQuery;
```

Comment:
You can fold this into the function signature, but don't do it if it looks less readable.